units/provisioning_convert_host_to_legacy.go

```diff
@@ -0,0 +1,231 @@
+			}
+			grip.Error(message.WrapError(j.Error(), message.Fields{
+				"message": "failed to convert host to legacy provisioning but will retry",
+				"host":    j.host.Id,
+				"distro":  j.host.Distro.Id,
+				"job":     j.ID(),
+			}))
+		}
+		grip.Error(message.WrapError(j.host.SetReprovisioningLocked(false), message.Fields{
+			"message": "could not clear host reprovisioning lock",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+	}()
+
+	// Lock the provisioning state to prevent other provisioning jobs from
+	// running.
+	if err := j.host.SetReprovisioningLockedAtomically(true); err != nil {
+		grip.Info(message.WrapError(err, message.Fields{
+			"message": "reprovisioning job currently in progress, returning from job",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+		return
+	}
+
+	if err := j.host.UpdateLastCommunicated(); err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "could not update host communication time",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+	}
+
+	settings := j.env.Settings()
+	sshOpts, err := j.host.GetSSHOptions(settings)
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "could not get SSH options",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+		j.AddError(err)
+		return
+	}
+	// This is a best-effort attempt to uninstall Jasper, but it will silently
+	// fail to uninstall Jasper if the Jasper binary path does not match its
+	// actual path on the remote host.
+	if logs, err := j.host.RunSSHCommandLiterally(ctx, fmt.Sprintf("[ -a \"%s\" ] && %s", j.host.JasperBinaryFilePath(settings.HostJasper), j.host.QuietUninstallJasperCommand(settings.HostJasper)), sshOpts); err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "could not uninstall Jasper service",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+			"logs":    logs,
+		}))
+		j.AddError(err)
+		return
+	}
+
+	if err := j.host.DeleteJasperCredentials(ctx, j.env); err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "could not delete Jasper credentials",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+		j.AddError(err)
+		return
+	}
+
+	if err := j.host.MarkAsReprovisioned(); err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "could not mark host as provisioned",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+		j.AddError(err)
+		return
+	}
+
+	event.LogHostConvertedProvisioning(j.host.Id, j.host.Distro.BootstrapSettings.Method)
+
+	grip.Info(message.Fields{
+		"message": "successfully converted host from non-legacy to legacy provisioning",
+		"host":    j.host.Id,
+		"distro":  j.host.Distro.Id,
+		"job":     j.ID(),
+	})
```

Comment:
maybe also worth dropping this?

Reply:
I do this mostly because the provisioning setup host job and agent deploy jobs do this. They are also significantly more log-heavy so I don't think this is a problem given that it runs so rarely.